### DISCUSSION TITLE
GeojsonLayer with TerrainLayer

### DISCUSSION THREAD

**Discussion by zogs** [visgl/deck.gl#5457] [Ideas]
Hello all,

I'm using DeckGL in a project to show vineyard in France. For that, TerrainLayer is usefull to visualize hillside of the parcels, and GeojsonLayer to show the contours. But to do that i need to extrude to geojson by a high amount to overlap the terrain layer and it does not render great for certains cases.
I am wondering if there is a way to get the geojson polygons non-extruded following the 3d terrain ? What would be the work to make for such feature ? I'm guessing it's non-trivial ...?

**Comment by kylebarron** [comment]
I think this is on our eventual roadmap, but it is non trivial, especially to do performantly on the GPU. I toyed around with this idea on the CPU last year, and the project might be useful for reference, but no promises that it works: https://github.com/kylebarron/snap-to-tin

**Reply by alienatorZ** [reply]
I would also like to see this feature.
